app/scripts/actions/index.js

```diff
@@ -264,6 +297,71 @@ export const clearCollectionsSearch = () => ({ type: CLEAR_COLLECTIONS_SEARCH })
 export const filterCollections = (param) => ({ type: FILTER_COLLECTIONS, param: param });
 export const clearCollectionsFilter = (paramKey) => ({ type: CLEAR_COLLECTIONS_FILTER, paramKey: paramKey });
 
+export const cumulusInstanceMetadata = () => wrapRequest(null, get, 'instanceMeta', ADD_CMR);
+
+/**
+ * Iterates over each collection in the application collections state dispatching the
+ * action to add the MMT link to the its state.
+ * @returns {function} anonymous redux-thunk function.
+ */
+export const getMMTLinks = () => {
+  return (dispatch, getState) => {
+    const { data } = getState().collections.list;
+    data.forEach((collection) => {
+      getMMTLinkFromCmr(collection, getState)
+        .then((url) => {
+          const action = {
+            type: ADD_MMTLINK,
+            data: { name: collection.name, version: collection.version, url: url }
+          };
+          dispatch(action);
+        })
+        .catch((error) => console.error(error));
+    });
+  };
+};
+
+/**
+ *
+ * @param {Object} collection - application collections item.
+ * @param {function} getState - redux-thunk callback.
+ * @returns {Promise<string>} - Promise for a Metadata Management Toolkit (MMT) Link
+ *                              to the input collection or null if it doesn't exist.
+ */
+export const getMMTLinkFromCmr = (collection, getState) => {
+  const {cmrProvider, cmrEnvironment} = getState().cumulusInstance;
```

Comment:
We should probably have an early return after this line if `cmrProvider` or `cmrEnvironment` are not set